### DISCUSSION TITLE
Impact of HashiCorps switch to BSL

### DISCUSSION THREAD

**Discussion by SebastianBezold** [eclipse-tractusx/sig-infra#233] [General]
# What happened

On Aug 10, 2023 HashiCorp announced a transition from the Mozilla Public License v2.0 (MPL 2.0) to the [Business Source License](https://www.hashicorp.com/bsl) (BSL, or BUSL) v1.1 for future releases of all products and several libraries. 

HashiCorp APIs, SDKs, and almost all other libraries will remain MPL 2.0.

## Infos about the [Business Source License](https://www.hashicorp.com/bsl) (BSL, or BUSL) v1.1:

- it is not an open source license
- see [SPDEX Entry](https://spdx.org/licenses/BUSL-1.1.html)
- Statement from the [Apache Foundation](https://www.apache.org/legal/resolved.html#category-x) 

## From the [HashiCorp FAQ](https://www.hashicorp.com/license-faq#products-covered-by-bsl):

The latest releases under MPL are: 

- Terraform 1.5.5, Packer 1.9.2, 
- Vault 1.14.1, 
- Boundary 0.13.1, 
- Consul 1.16.1, 
- Nomad 1.6.1, 
- Waypoint 0.11.4,  
- Vagrant 2.3.7, which is MIT licensed. 

Security fixes will be backported under MPL 2.0 through December 31, 2023.

# Potential impact on Tractus-X

We have to investigate and need help from the product teams, please provide input:
- Are you using one of the tools mentioned above?
- If yes, how / in which context?
- Check your DEPENDENCIES file(s), if there are any HashiCorp entries, please contact FOSS Team asap (via a comment in this issue) 

**If you are using one of the tool do NOT upgrade to higher versions as mentioned above!**

**Comment by evegufy** [comment]
Portal product is not affected (no HashiCorp dependencies)

**Comment by drcgjung** [comment]
Knowledge Agents -> Agent Plane Hashicorp Flavour depends on Tractus-X EDC Data Plane Hashicorp Flavour.
There is no additional code / secrets added to the Tractus-X EDC code/secrets.

**Comment by tom-rm-meyer-ISST** [comment]
puris is affected indirectly. It has a local deployment including the Hashicorp Vault in Version 1.13.3 to spin up EDCs. The application itself does not rely on the vault.

**Comment by florianrusch-zf** [comment]
The EDC is using HashiCorp Vault for storing secrets. Therefor it contains a Client implementation: https://github.com/eclipse-edc/Connector/tree/main/extensions/common/vault/vault-hashicorp

**Comment by hzierer** [comment]
Last Thursday the [OpenTofu](https://github.com/opentofu/opentofu/) fork made their [first release](https://github.com/opentofu/opentofu/releases/tag/v1.6.0-alpha2). As this is a drop-in replacement I think we should give it a try. Unfortunately there are no official Docker images yet - but I think this only a matter of days.

(PS: As this fork is supported by the [Linux Foundation](https://www.linuxfoundation.org/press/announcing-opentofu) I doubt it will go away anytime soon)
